### PR TITLE
Implementing post_process_triangulation

### PR DESCRIPTION
**Pull request by ESeNonFossiIo**

### CODE REVIEW

File: include/interface.h [hunk 1]
```diff
@@ -98,6 +98,12 @@ class Interface : public ParsedFiniteElement<dim,spacedim>
   }
 
   /**
+   * This function is used to modify triangulation using boundary_id or manifold_id.
+   * In the case a signal is required, this is the function to modify.
+   */
+  virtual void post_process_triangulation(const Triangulation<dim, spacedim> &tria) const {}
```
Comment (luca-heltai) [inline include/interface.h]: The argument CANNOT be constant...

Comment (luca-heltai) [inline include/interface.h]: Move the default implementation in the cc file.

Comment (luca-heltai) [inline include/interface.h]: `postprocess_newly_created_triangulation` ?

This name would be much more specific...

File: source/interface.cc [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+{
+  auto &sol = fe_cache.get_current_independent_local_dofs("solution", alpha);
+  auto &sol_dot = fe_cache.get_current_independent_local_dofs("solution_dot", alpha);
+
+  for (unsigned int i=0; i<sol.size(); ++i)
+    sol_dot[i] = (alpha.val().val()*sol[i]) + (sol_dot[i].val().val() - alpha.val().val()*sol[i].val().val());
+}
+
+
+template <int dim, int spacedim, int n_components, typename LAC>
```
Comment (ESeNonFossiIo) [inline source/interface.cc]: I have some problem with the explicit instantiation of this method. @asartori86 and @luca-heltai any suggestion?

**Pull request merged**